var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -36,14 +36,16 @@ class Magma(CMakePackage, CudaPackage):
     depends_on('blas')
     depends_on('lapack')
     depends_on('cuda@8:', when='@2.5.1:')  # See PR #14471
```

Comment:
Should this line be a `conflicts` as well? Otherwise, older versions of this package do not depend on cuda at all.

Reply:
They would pick up CUDA from the `CudaPackage` mixin.
```
class Magma(CMakePackage, CudaPackage):  
```